contracts/treasury/GovernorCountingOverridable.sol

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    /**
+     * @notice Handles vote overrides that delegators can make to their
+     * corresponding delegated transcoder votes. Usually only the transcoders
+     * vote on proposals, but any delegator can change their part of the vote.
+     * This tracks past votes and deduction on separate mappings in order to
+     * calculate the effective voting power of each vote.
+     * @param _proposalId ID of the proposal being voted on
+     * @param _tally struct where the vote totals are tallied on
+     * @param _voter struct where the specific voter state is tracked
+     * @param _account current user making a vote
+     * @param _weight voting weight of the user making the vote
+     */
+    function _handleVoteOverrides(
+        uint256 _proposalId,
+        ProposalTally storage _tally,
+        ProposalVoterState storage _voter,
+        address _account,
+        uint256 _weight
+    ) internal returns (uint256) {
+        uint256 timepoint = proposalSnapshot(_proposalId);
+        address delegate = votes().delegatedAt(_account, timepoint);
+
+        bool isTranscoder = _account == delegate;
+        if (isTranscoder) {
+            // deduce weight from any previous delegators for this transcoder to
+            // make a vote
+            return _weight - _voter.deductions;
+        }
+
+        // this is a delegator, so add a deduction to the delegated transcoder
+        ProposalVoterState storage delegateVoter = _tally.voters[delegate];
+        delegateVoter.deductions += _weight;
+
+        if (delegateVoter.hasVoted) {
+            // this is a delegator overriding its delegated transcoder vote,
+            // we need to update the current totals to move the weight of
+            // the delegator vote to the right outcome.
+            VoteType delegateSupport = delegateVoter.support;
+
+            if (delegateSupport == VoteType.Against) {
+                _tally.againstVotes -= _weight;
+            } else if (delegateSupport == VoteType.For) {
+                _tally.forVotes -= _weight;
+            } else {
+                assert(delegateSupport == VoteType.Abstain);
+                _tally.abstainVotes -= _weight;
+            }
+        }
+
+        return _weight;
+    }
+
+    /**
+     * @dev Implement in inheriting contract to provide the voting power provider.
+     */
+    function votes() public view virtual returns (IVotes);
+
+    /**
+     * @dev Implement in inheriting contract to provide quota value to use to decide proposal success.
+     * @return quota value as a MathUtils percentage value (e.g. 6 decimal places).
+     */
+    function quota() public view virtual returns (uint256);
+
+    /**
+     * @dev This empty reserved space is put in place to allow future versions to add new
+     * variables without shifting down storage in the inheritance chain.
+     * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+     */
+    uint256[50] private __gap;
```

Comment:
Any reason for 50 vs 49 as is used here? https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/3d4c0d5741b131c231e558d7a6213392ab3672a5/contracts/governance/extensions/GovernorCountingSimpleUpgradeable.sol#L112

Reply:
In the lack of further documentation about standards for this apart from ["mind the gap"](https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps) [1], I looked around on OZ contracts and my first understanding was that they always started these gaps as `50` and then reduced it as new state variables were introduced. After your comment I took another look and realized that the idea (at least around OZ) is to always keep the contracts with a storage layout of 50 slots, reducing the gap depending on the amount of vars it has even on the first version of the contract.

So I'm changing this to `49` to have the whole contract have 50 slots size. Actually 48, since I'm adding a new variable for the `quota` now.

[1] I had to make that joke, but just before replying here I found [these docs](https://docs.openzeppelin.com/contracts/3.x/upgradeable#storage_gaps) which does mention the "50 slots" standard. It's still just an OZ convention anyway, but no harm in following it and better to keep the consistency.